src/js/components/AppListComponent.jsx

```diff
@@ -186,7 +198,11 @@ var AppListComponent = React.createClass({
 
     if (filterText != null && filterText !== "") {
       nodesSequence = nodesSequence
-        .filter(app => app.id.indexOf(filterText) !== -1);
+        .filter(app => {
+          app.filterScore = score(app.id, filterText);
```

Comment:
That won't work, however, if we plan on supporting manual overriding of the sorting. I would like, for example, to search for `phil` and get the ranked sorting. But if I then click on one of the sorting columns, (for example CPU), this should override the search rank. 
My solution is hacky so it's a no-go, so let's have a chat about a better solution that addresses all of that.